### PR TITLE
Upgrade to using React hooks

### PR DESCRIPTION
**Pull request by stephtaylor**
- Update to use React hooks
- Fix auto doc generation (was relying on React.Component to be present)
- Add info about leveraging the Context component

Fixes:
- #44 
- #41

### CODE REVIEW

File: docs/components/ApplePayButton.md [hunk 1]
```diff
@@ -4,7 +4,7 @@ title:ApplePayButton
...
 Renders an Apple Pay button to use in the Square Payment Form, pre-styled to meet Apple's Human Interface Guidelines.
 ## Props
-|Name|Type|Description|
-|---|---|---|
-|loadingView|React.ReactNode|Placeholder view when the Apple Pay is being initialized|
-|unavailableView|React.ReactNode|Placeholder view when Apple Pay is not available|
\ No newline at end of file
+|Name|Type|Description|Default Value|
+|---|---|---|---|
+|loadingView|ReactReactNode|Placeholder view when the Apple Pay is being initialized||
```
Comment (nlitwin) [inline docs/components/ApplePayButton.md]: Should this one stay the same? `React.ReactNode`

File: src/components/Context.tsx [hunk 2]
```diff
@@ -1,18 +1,31 @@
...
 export interface ContextInterface {
-  formId?: string
-  applePayState?: string
-  googlePayState?: string
-  masterpassState?: string
-  onCreateNonce?: (event: React.MouseEvent) => {}
+  /** Unique form ID */
+  formId?: string;
+  /** Apple pay state (loading, unavailable, ready) */
+  applePayState?: string;
```
Comment (nlitwin) [inline src/components/Context.tsx]: Could we do `applePayState?: 'loading' | 'unavailable' | 'ready';` for these?

File: docs/components/ApplePayButton.md (hunk 1 above)
Comment (stephtaylor) [reply docs/components/ApplePayButton.md]: Done!

File: src/components/Context.tsx (hunk 2 above)
Comment (stephtaylor) [reply src/components/Context.tsx]: Done!

File: docs/charging.md [hunk 3]
```diff
@@ -1,5 +1,5 @@
 ---
-id: charging
+id: payments
```
Comment (maxbeatty) [inline docs/charging.md]: if this `id` is changing, should the file name?

Comment (stephtaylor) [reply docs/charging.md]: 👍

**Pull request merged**